contracts/RewardDelegate.sol

```diff
@@ -0,0 +1,79 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+pragma solidity 0.7.6;
+
+import { Address } from "@openzeppelin/contracts/utils/Address.sol";
+import { SafeMath } from "@openzeppelin/contracts/math/SafeMath.sol";
+import { IRewardDelegate } from "./interface/IRewardDelegate.sol";
+
+contract RewardDelegate is IRewardDelegate {
+    using Address for address;
+    using SafeMath for uint256;
+
+    // truster => beneficiaryCandidate
+    mapping(address => address) internal _beneficiaryCandidateMap;
+    // truster => beneficiary
+    mapping(address => address) internal _beneficiaryMap;
+    // beneficiary => how many trusters delegate to this beneficiary
+    mapping(address => uint256) internal _trusterCountMap;
```

Comment:
definition question: when no one set `alice` as their beneficiary, should `_trusterCountMap[alice] == 0 or 1`?

Reply:
if no one set to alice, `_trusterCountMap[alice]` will be 0